Testscripts/Linux/validate-da.sh

```diff
@@ -22,148 +22,165 @@ set -x
 # Get distro information
 GetDistro
 
-dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
-supported_distro_list=(redhat centos suse debian ubuntu)
+readonly dir_list=(/opt/microsoft/dependency-agent /var/opt/microsoft/dependency-agent /etc/opt/microsoft/dependency-agent /lib/microsoft-dependency-agent /etc/init.d/microsoft-dependency-agent)
+readonly supported_distro_list=(redhat centos suse debian ubuntu)
 
 readonly da_pid_file="/etc/opt/microsoft/dependency-agent/config/DA_PID"
+readonly uninstaller="/opt/microsoft/dependency-agent/uninstall"
 
 function test_case_cleanup(){
-    /opt/microsoft/dependency-agent/uninstall
+    [ -f "$uninstaller" ] && eval $uninstaller
 
-    for dir in "${dir_list[@]}"
-    do
-        rm -rf $dir
-    done
+    rm -rf "${dir_list[@]}"
 
-    kill $watch_pid
+    # This can be a NOP if the test exits before calling generate_network_activity
+    kill $network_activity_generator_pid
 
     LogMsg "Finished cleaning up test cases"
 }
 
 # Calls the cleanup function when interupted or exited
-trap test_case_cleanup SIGINT EXIT
+trap test_case_cleanup SIGINT EXIT SIGHUP SIGQUIT SIGTERM
+
+function fail_test() {
+    LogErr "$*"
+    SetTestStateFailed
+    exit 0
+}
 
-function verify_directory() {
-    failed_assertions=0
+function verify_directory_exists() {
+    failed_assertions=false
     for dir in "${dir_list[@]}"
     do
-        [ -d $dir ] && LogErr "Directory $dir exists." && SetTestStateFailed && failed_assertions=$[[failed_assertions+1]]
+        [ -d $dir ] && LogErr "Directory $dir exists." && failed_assertions=true
```

Comment:
If `LogErr` should return a non-zero result, then `failed_assertions` will not be set.

Recommendation: Keep it simple. Don't use the `&&` unless there really is a dependency between the left and right operands.

```
if [ -d "$dir" ]; then
    LogErr "Directory $dir exists."
    failed_assertions=true
fi
```

Reply:
Done